docs/DeveloperGuide.md

```diff
@@ -1,38 +1,201 @@
+  - [UI class](#ui-class)
+  - [DataStorage class](#datastorage-class)
+    - [loadChassis() method](#loadchassis-method)
+  - [Parser class](#parser-class)
+  - [PC Class Architecture](#pc-class-architecture)
+  - [Command class](#command-class)
+    - [`list [COMPONENT]` Command (builder mode)](#list-component-command-builder-mode)
+- [Appendix: Requirements](#appendix-requirements)
+  - [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+  - [User Stories](#user-stories)
+  - [Non-Functional Requirements](#non-functional-requirements)
+  - [Glossary](#glossary)
+  - [Instructions for manual testing](#instructions-for-manual-testing)
+
+
+## General Overview
+below is the overall architecture diagram for how BigPP works.
+
+![Architecture Diagram](uml-pictures/ArchitectureDiagram.png)
+
+The program will first load the `UserJson` and files in the `Resources` folder to populate its internal memory of `PCLists` and `ComponentLists`. This will be stored in its `DataStorage`. The `User`'s interaction with the `UI` will be `parsed` into a `command` which would update the `DataStorage` and eventually update the `Menu` which is displayed back to the `User`. This would continue until the `User` exits the program, which would result in the data stored in `DataStorage` being saved into the `UserJson`.
 
-## Acknowledgements
+## Design & implementation
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### UI class
+Below is the Class diagram for the UI class
 
-## Design & implementation
+![UI Class Diagram](uml-pictures/UIClassDiagram.png)
+
+The `UI` Class serves as the interface between the user and BigPP. It is responsible for the following key actions
+
+1. **Initialization**: Upon start up of BigPP, the instance of `DataStorage` is parsed into the `updateUI` method of `UI` to give it the ability to access resources for BigPP. `UIState` is set to be `PCViewer`.
+   
+2. **Functionality**
+  - Shows welcome message to the user
+  - Gets input from the user
+  - Sets the `UIState` of BigPP depending on user command
+  - Updates the user interface to show the menu depending on the `UIState`
+  - Clears the terminal after every user command
+  - Shows feedback to the user after every user command
+
+3. **Behavior**: The `UI` Class acts as the intermediary between the user and BigPP. `UIState` of `UI` can change between `PCViewer` and `PCBuilder` which would result in different commands available to the user. This makes the possible commands that can be given clear to the user.
+
+In summary, UI gives crucial feedback to the user such as the menu of possible commands based on the current menu and results of a given command.
+
+### DataStorage class
+Below is the Class diagram for the DataStorage class:
+
+![DataStorage Class Diagram](uml-pictures/DataStorageClass.png)
```

Comment:
Do the PATHs in the DataStorage Class need to be included in this diagram? If the PATHs are final, could they be removed to make the diagram more readable?